ansible/roles/data-products-deploy/templates/submit-script.j2

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env bash
+
+export SPARK_HOME={{ analytics.home }}/spark-{{ spark_version }}-bin-hadoop2.7
+export MODELS_HOME={{ analytics.home }}/models-{{ model_version }}
+export DP_LOGS={{ analytics.home }}/logs/data-products
+export KAFKA_HOME={{ analytics.home }}/kafka_2.11-0.10.1.0
+
+## Job to run daily
+cd /mount/data/analytics/scripts
+source model-config.sh
+today=$(date "+%Y-%m-%d")
+
+## job broker-list and kafka-topic
+zookeeper={{ zookeeper }}
+brokerList={{ brokerlist }}
+job_topic={{ analytics_job_queue_topic }}
+unique_topic={{ env }}.telemetry.unique
+
+while :; do
+    case $1 in
+        -c|--job)   shift 
+                    job="$1"            
+        ;;
+        -m|--mode)  shift 
+                    type="$1"            
+        ;;
+        -p|--partitions)    shift 
+                            partitions=$1          
+        ;;
+        -s|--parallelisation)   shift 
+                                parallelisation=$1            
+        ;;
+        -sd|--startDate)   shift 
+                            start_date=$1            
+        ;;
+        -ed|--endDate)     shift 
+                            end_date=$1            
+        ;;
+        -h|--sparkMaster)   shift 
+                            sparkMaster=$1            
+        ;;
+        *) break
+    esac
+    shift
+done
+
+if [ -z "$sparkMaster" ]; then sparkMaster="local[*]"; else sparkMaster="$sparkMaster"; fi
+
+if [ "$type" = "via-partition" ]; then
+    # get partitions from kafka metadata
+    if [ -z "$partitions" ]; then partitions=`$KAFKA_HOME/bin/kafka-topics.sh --zookeeper $zookeeper --describe --topic $unique_topic   | awk 'NR==1{print $2}' |  awk -F":" '{print $2}'`; else partitions=$partitions; fi
+    endPartitions=`expr $partitions - 1`
+    if [ -z "$parallelisation" ]; then parallelisation=1; else parallelisation=$parallelisation; fi
+    # add partitions to config and start jobs
+    for i in $(seq 0 $parallelisation $endPartitions)
+        do 
+            # add partitions to config
+            partitionString="\"delta\":0,\"partitions\":[$(seq -s , $i `expr $i + $parallelisation - 1`)]"
+            if [ -z "$start_date" ]; then
```

Comment:
I think we should check the start_date condition and initialize the start_date and end_date accordingly. This will allow us to have the spark-submit code only once.

Reply:
refactored to use spark-submit only once